subxt/src/custom_values/custom_values_client.rs

```diff
@@ -30,11 +30,15 @@ impl<T: Config, Client: OfflineClientT<T>> CustomValuesClient<T, Client> {
         &self,
         address: &Address,
     ) -> Result<Address::Target, Error> {
+        // 1. Validate custom value shape if hash given:
+        self.validate(address)?;
+
+        // 2. Attempt to decode custom value:
         let metadata = self.client.metadata();
-        let custom_value = metadata
-            .custom()
+        let custom = metadata.custom();
+        let custom_value = custom
             .get(address.name())
-            .ok_or_else(|| MetadataError::CustomValueNameNotFound(address.name().to_string()))?;
+            .expect("custom value is present, checked above; qed");
```

Comment:
It's only checked above if validation is enabled (ie static values with validation passed). If somebody passes some arbitrary `str` address it'll fail. So best to stick to the error you were returning before :)